web/src/filter.ts

```diff
@@ -206,6 +206,19 @@ function message_matches_search_term(message: Message, operator: string, operand
             return message.stream_id.toString() === operand;
         }
 
+        case "channels":
+            if (message.type !== "stream") {
+                return false;
+            }
+            switch (operand) {
+                case "public":
+                    return !stream_data.is_stream_archived(message.stream_id);
```

Comment:
This is not how you check if a channel is public. 

Are you manually testing your work and self-reviewing the code carefully? It's a waste of reviewers' time to be pestering people for review when you're not doing your own basic diligence.